compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -1035,7 +1035,7 @@ static void PrintService(const ServiceDescriptor* service,
       "@$Generated$(\n"
       "    value = \"by gRPC proto compiler$grpc_version$\",\n"
       "    comments = \"Source: $file_name$\")\n"
-      "public class $service_class_name$ {\n\n");
+      "public final class $service_class_name$ {\n\n");
```

Comment:
I'm OK with MethodHandlers being final, but why should `$service_class_name$` be final?